src/App.jsx

```diff
@@ -14,45 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_TYPE_ALPHABET = 'Sort alphabetically';
+const SORT_TYPE_LENGTH = 'Sort by length';
+
+function getPreparedGoods(goods, { sortType, isReverse }) {
+  let preparedGoods = [...goods];
+
+  if (isReverse === 1) {
+    preparedGoods = preparedGoods
+      .sort((good1, good2) => preparedGoods.indexOf(good2)
+        - preparedGoods.indexOf(good1));
+  }
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SORT_TYPE_ALPHABET: {
+        if (isReverse === 0) {
+          return good1.localeCompare(good2);
+        }
+
+        return good2.localeCompare(good1);
+      }
+
+      case SORT_TYPE_LENGTH: {
+        if (isReverse === 0) {
+          return good1.length - good2.length;
+        }
+
+        return good2.length - good1.length;
+      }
+
+      default:
+        return 0;
+    }
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(0);
```

Comment:
```suggestion
  const [isReverse, setIsReverse] = useState(false);
```
it can be the boolean state